src/pyscaffoldext/custom_extension/custom_extension.py

```diff
@@ -57,3 +67,13 @@ def add_entry_point(self, struct, opts):
     def get_class_name_from_opts(opts):
         pkg_name = opts["package"]
         return "".join(map(str.capitalize, pkg_name.split("_")))
+
+
+def set_pyscaffoldext_namespace(struct, opts):
+    namespace_parameter = opts.get("namespace", None)
+    namespace_with_pyscaffoldext = ".".join([PYSCAFFOLDEXT_NS,
+                                             namespace_parameter]) \
+        if namespace_parameter else \
+        PYSCAFFOLDEXT_NS
```

Comment:
This is quite hard to read, what about:
```python
ns_pyscaffold = PYSCAFFOLDEXT_NS
ns_param = opts.get("namespace", None)
if ns_param:  # nest provided namespace within pyscaffoldext
   ns_pyscaffold = ".".join([ns_pyscaffold, ns_param])
  
opts["namespace"] = prepare_namespace(ns_pyscaffold)
```

The trick is that you don't need extra long variable names if they are used only in a short scope.